megatron/optimizer/distrib_optimizer.py

```diff
@@ -385,10 +385,14 @@ def __init__(self, optimizer, clip_grad, log_num_zeros_in_grad,
         #   storage & have their own dtype. This is safe because the param
         #   dtype size is always <= grad dtype size.
         self.param_buffers = []
+        untyped_attr = "untyped"
+        if not hasattr(torch.Storage, untyped_attr):
```

Comment:
Maybe check `torch.TypedStorage` (torch < 1.13) or `torch_TypedStorage` (torch >= 1.13) are better? considering  `torch.Storage` is only an alias for the default storage data type. Typically, it resolves to `torch.FloatStorage`, but could also correspond to other types. [reference](https://pytorch.org/docs/1.12/storage.html)

Anyway, there are still differences between TypedStorage and _TypedStorage. Maybe directly check pytorch version is more robust?
```python
if torch.__version__ >= torch.torch_version.TorchVersion('1.13'):
    pass
```

Reply:
@drcege It seems better. Actually, I think we can make an alias in `megatron/arguments.py`, either https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/arguments.py#L181 or https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/arguments.py#L298 (since Megatron retrieves the version info at here).

+) Furthermore, it seems we can replace https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/arguments.py#L299 with `if torch.__version__ < torch.torch_version.TorchVersion('1.11'):`, instead of parsing `TORCH_MAJOR` and `TORCH_MINOR`

Reply:
@ktaebum  I personally add these lines to the top of [distrib_optimizer.py](https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/optimizer/distrib_optimizer.py#L17), because it is more like a  hack and should not be exposed to end users.